File: wordcount/nodejs/reducer.js
#!/usr/bin/env node
var byline = require('byline');

var lines = {};
var stream = byline(process.stdin, {encoding:'utf8'});
stream.on('data', function(d) {
  //  Should be in (key, count) touples
  var pair = d.split(/\s+/);
  if (!lines[pair[0]]) {
    lines[pair[0]] = pair[1]; 
  } else {
    lines[pair[0]] += pair[1]; 
  }
});
stream.on('end', function() {
  for (var k in lines) {
    console.log(k+" "+lines[k]);
  }
});

File: reducer.pl
#!/usr/bin/perl

my $currentWord = "", $total = 0;
my $word = "";

while ($line = <>) {
  chop($line);
  my $count = 0;
  
  ($word, $count) = split(' ', $line);
  if ($word eq $currentWord) {
    #print "$word eq $currentWord\n";
    $total += $count;
  } else {
    if ($currentWord ne "") {
      #print "$currentWord != ''\n";
      print "$currentWord $total\n";
    }
    $currentWord = $word;
    $total = $count;
  }
}

if ($currentWord eq $word) {
  #print "$currentWord == $word ''\n";
  print "$currentWord $total\n";
}

